client/java/src/main/java/io/openlineage/client/circuitBreaker/TaskQueueCircuitBreaker.java

```diff
@@ -46,9 +46,7 @@ public TaskQueueCircuitBreaker(@NonNull TaskQueueCircuitBreakerConfig config) {
     this.blockingTimeInSeconds = config.getBlockingTimeInSeconds();
     this.shutdownTimeoutSeconds = config.getShutdownTimeoutSeconds();
     eventQueue = new ArrayBlockingQueue<>(config.getQueueSize());
-    eventProcessingExecutor =
-        new ThreadPoolExecutor(
-            config.getThreadCount(), config.getThreadCount(), 60L, TimeUnit.SECONDS, eventQueue);
+    eventProcessingExecutor = OpenLineageClientUtils.getOrCreateExecutor();
```

Comment:
@pawel-big-lebowski have you left the threadpoolexecutor here on purpose? 

If yes, then we need non-static thread factory then